lib/esshorten.js

```diff
@@ -74,9 +74,10 @@
             return true;
         }
 
-        generateName(tip) {
+        generateName(tip, prefix) {
             do {
-                tip = utility.generateNextName(tip);
+                tip = tip.replace(prefix, '');
```

Comment:
The first time the `do...while` executes `tip` is not prefixed yet so I guess slicing the string won't work as expected. Am I missing something?